scylla/tests/integration/batch_statements.rs

```diff
@@ -0,0 +1,403 @@
+    session.use_keyspace(&ks, false).await.unwrap();
+    session
+        .ddl(format!(
+            "CREATE TABLE IF NOT EXISTS {} (k0 text, k1 int, v int, PRIMARY KEY (k0, k1))",
+            table_name
+        ))
+        .await
+        .unwrap();
+
+    (session, ks)
+}
+
+async fn create_counter_tables(session: &Session, ks: &str) {
+    for &table in ["counter1", "counter2", "counter3"].iter() {
+        session
+            .ddl(format!(
+                "CREATE TABLE {}.{} (k0 text PRIMARY KEY, c counter)",
+                ks, table
+            ))
+            .await
+            .unwrap();
+    }
+}
+
+async fn verify_batch_insert(session: &Session, keyspace: &str, test_name: &str, count: usize) {
+    let select_query = format!(
+        "SELECT k0, k1, v FROM {}.{} WHERE k0 = ?",
+        keyspace, test_name
+    );
+    let query_result = session
+        .query_unpaged(select_query, (test_name,))
+        .await
+        .unwrap()
+        .into_rows_result()
+        .unwrap();
+    let rows: Vec<(String, i32, i32)> = query_result
+        .rows::<(String, i32, i32)>()
+        .unwrap()
+        .map(|r| r.unwrap())
+        .collect();
+    assert_eq!(rows.len(), count);
+    for (k0, k1, v) in rows {
+        assert_eq!(k0, test_name);
+        assert_eq!(v, k1 + 1);
+    }
+}
+
+async fn prepare_insert_statement(session: &Session, ks: &str, table: &str) -> PreparedStatement {
+    let query_str = format!("INSERT INTO {}.{} (k0, k1, v) VALUES (?, ?, ?)", ks, table);
+    session.prepare(Query::new(query_str)).await.unwrap()
+}
+
+#[tokio::test]
+async fn test_batch_of_simple_statements() {
+    setup_tracing();
+    let test_name = String::from("test_batch_simple_statements");
+    let (session, ks) = create_test_session(&test_name).await;
+
+    let mut batch = Batch::new(BatchType::Unlogged);
+    for i in 0..BATCH_COUNT {
+        let simple_statement = Query::new(format!(
+            "INSERT INTO {}.{} (k0, k1, v) VALUES ('{}', {}, {})",
+            ks,
+            &test_name,
+            &test_name,
+            i,
+            i + 1
+        ));
+        batch.append_statement(simple_statement);
+    }
+    session.batch(&batch, vec![(); BATCH_COUNT]).await.unwrap();
+    verify_batch_insert(&session, &ks, &test_name, BATCH_COUNT).await;
+}
+
+#[tokio::test]
+async fn test_batch_of_bound_statements() {
+    setup_tracing();
+    let test_name = String::from("test_batch_bound_statements");
+    let (session, ks) = create_test_session(&test_name).await;
+
+    let prepared = prepare_insert_statement(&session, &ks, &test_name).await;
+    let mut batch = Batch::new(BatchType::Unlogged);
+    let mut batch_values: Vec<Vec<CqlValue>> = Vec::with_capacity(BATCH_COUNT);
+    for i in 0..BATCH_COUNT as i32 {
+        batch.append_statement(prepared.clone());
+        batch_values.push(vec![
+            CqlValue::Text(test_name.clone()),
+            CqlValue::Int(i),
+            CqlValue::Int(i + 1),
+        ]);
+    }
```

Comment:
You can use `Vec<(String, i32, i32)>` for `batch_values` - no need to go through another vector, and CqlValue.